### ISSUE TITLE
Null analysis seems to lose non-null information for delegates captured in lambdas

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#31584]
**Version Used**: VS 16.0 Preview 1

**Steps to Reproduce**:

1. Copy the code into a C# 8.0 project with nullable reference types enabled.

```c#
void M<T>(Func<T, bool>? predicate)
{
    var items = Enumerable.Empty<T>();
    if (predicate != null)
        items = items.Where(x => predicate(x));
}
```

**Expected Behavior**: No warning about null dereference is reported, as `predicate` was checked to be not null.

**Actual Behavior**: Warning CS8602 ("Possible deference of a null reference") is reported on the delegate invocation.

I've asked two colleagues about this, and both said they would expect `predicate` to be non-null after the null check, including inside the lambda.
